### PR TITLE
Core migration sheets - Reading from Sheets.

### PR DESCRIPTION
**Pull request by zradojcicExecom**

### CODE REVIEW

File: ExeBite.Sheets/ExeBite.Sheets.Common/Constants.cs [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+        /// Name of the Pod Lipom Restaurant
+        /// </summary>
+        public const string POD_LIPOM_NAME = "Pod Lipom";
+
+        /// <summary>
+        /// Name of the Teglas restaurant
+        /// </summary>
+        public const string TEGLAS_NAME = "Tegla's";
+
+        public const string NO_CATEGORY = "Unspecified";
```
Comment (gmocelj) [inline ExeBite.Sheets/ExeBite.Sheets.Common/Constants.cs]: Add summary for what is used

File: ExeBite.Sheets/ExeBite.Sheets.Common/Food.cs [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+        public string Subcategory { get; private set; }
+
+        /// <summary>
+        /// Standard constructor.
+        /// Protected so that only child classes can be instantiated.
+        /// </summary>
+        /// <param name="name"></param>
+        /// <param name="price"></param>
+        /// <param name="restaurant"></param>
+        /// <param name="subcategory"></param>
```
Comment (gmocelj) [inline ExeBite.Sheets/ExeBite.Sheets.Common/Food.cs]: Add description of params

File: ExeBite.Sheets/ExeBite.Sheets.Hedone/HedoneReader.cs [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+                    DateTime parsedDate = result.Value;
+
+                    if (providedDate.Year != parsedDate.Year
+                        || providedDate.Month != parsedDate.Month)
+                        break;
+
+                    if (providedDate.Date.Equals(parsedDate.Date))
+                        return Result<MergedRegion>.Success(merge);
+                }
+                Thread.Sleep(50);
```
Comment (gmocelj) [inline ExeBite.Sheets/ExeBite.Sheets.Hedone/HedoneReader.cs]: Write the comment why is this needed

File: ExeBite.Sheets/ExeBite.Sheets.Hedone/DataExtractor.cs [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+        /// Used to extract standing food offer for the restaurant.
+        /// </summary>
+        /// <param name="ranges"></param>
+        /// <returns></returns>
+        public static List<FoodItem> ExtractFoodItems(ValueRange ranges)
+        {
+            var foundFood = new List<FoodItem>();
+            foreach (var row in ranges.Values)
+            {
+                if (row.Count < 4)
```
Comment (gmocelj) [inline ExeBite.Sheets/ExeBite.Sheets.Hedone/DataExtractor.cs]: Can this be written as Skip(4) in foreach part,  instead of <4?

File: ExeBite.Sheets/ExeBite.Sheets.Index/DataExtractor.cs [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+        /// <param name="ranges"></param>
+        /// <returns></returns>
+        public static List<FoodItem> ExtractFoodItems(ValueRange ranges)
+        {
+            var foundFood = new List<FoodItem>();
+
+            string category = string.Empty;
+            foreach (var row in ranges.Values)
+            {
+                if (row.Count < 3)
```
Comment (gmocelj) [inline ExeBite.Sheets/ExeBite.Sheets.Index/DataExtractor.cs]: Same as above, check if this can be done as ranges.Values.Skip(3) in foreach?

File: ExeBite.Sheets/ExeBite.Sheets.PodLipom/PodLipomReader.cs [hunk 6]
```diff
@@ -0,0 +1,146 @@
...
+                    DateTime parsedDate = result.Value;
+
+                    if (providedDate.Year != parsedDate.Year
+                        || providedDate.Month != parsedDate.Month)
+                        break;
+
+                    if (providedDate.Date.Equals(parsedDate.Date))
+                        return Result<MergedRegion>.Success(merge);
+                }
+                Thread.Sleep(50);
```
Comment (gmocelj) [inline ExeBite.Sheets/ExeBite.Sheets.PodLipom/PodLipomReader.cs]: Add comment why is this needed. 

Also, maybe this time for the sleep can be in constants, so that if for some reason this 50ms is not enough to easy change on all places this

File: ExeBite.Sheets/ConsoleDebug/Program.cs [hunk 7]
```diff
@@ -19,21 +19,22 @@ static void Main(string[] args)
...
             var providedDate = DateTime.ParseExact("9/3/2018", "M/d/yyyy", CultureInfo.InvariantCulture);
 
-            //var hed = new HedoneReader(logger);
-            //var DailyOffers = hed.ReadDailyOffers(providedDate);
-            //var PerpetualOffers = hed.ReadFoodItems();
+            var teg = new TeglasReader(logger);
+            var DailyOffers = teg.ReadDailyOffers(providedDate);
+            var PerpetualOffers = teg.ReadFoodItems();
+
+            //var index = new IndexReader(logger);
```
Comment (gmocelj) [inline ExeBite.Sheets/ConsoleDebug/Program.cs]: Remove commented code

File: Exebite.JobScheduler/Unity/UnityConfig.cs [hunk 8]
```diff
@@ -49,9 +50,10 @@ public static void RegisterTypes(IUnityContainer container)
             container.RegisterType<IGoogleSheetService, GoogleSheetService>();
 
             container.RegisterType<IRestaurantConector, RestaurantConector>();
-            container.RegisterType<IHedoneConector, HedoneConector>();
-            container.RegisterType<ILipaConector, LipaConector>();
-            container.RegisterType<ITeglasConector, TeglasConector>();
+            container.RegisterType<ISheetsAPI, SheetsAPI>();
+            container.RegisterType<IHedoneConector, HedoneConector>(); // Probably deprecated
```
Comment (gmocelj) [inline Exebite.JobScheduler/Unity/UnityConfig.cs]: Remove this if is not in use anymore. Also, try to run exebite to see that something is not missed in unity configuration

File: ExeBite.Sheets/ExeBite.Sheets.Hedone/HedoneReader.cs (hunk 3 above)
Comment (zradojcicExecom) [reply ExeBite.Sheets/ExeBite.Sheets.Hedone/HedoneReader.cs]: Added comment and moved magic number to constants file.

File: ExeBite.Sheets/ConsoleDebug/Program.cs (hunk 7 above)
Comment (zradojcicExecom) [reply ExeBite.Sheets/ConsoleDebug/Program.cs]: Only used for testing.
